rllib/algorithms/ddpg/apex.py

```diff
@@ -7,29 +7,91 @@
 from ray.rllib.evaluation.worker_set import WorkerSet
 from ray.rllib.utils.annotations import override
 from ray.rllib.utils.typing import TrainerConfigDict
-from ray.util.iter import LocalIterator
 from ray.rllib.utils.typing import PartialTrainerConfigDict
 from ray.rllib.utils.typing import ResultDict
-from ray.rllib.utils.deprecation import DEPRECATED_VALUE
+from ray.rllib.utils.deprecation import Deprecated, DEPRECATED_VALUE
+from ray.util.iter import LocalIterator
+
+
+class ApexDDPGConfig(DDPGConfig):
+    """Defines a configuration class from which an ApexDDPGTrainer can be built.
+
+    Example:
+        >>> from ray.rllib.algorithms.ddpg.apex import ApexDDPGConfig
+        >>> config = ApexDDPGConfig().training(lr=0.01).resources(num_gpus=1)
+        >>> print(config.to_dict())
+        >>> # Build a Trainer object from the config and run one training iteration.
+        >>> trainer = config.build(env="Pendulum-v1")
+        >>> trainer.train()
+
+    Example:
+        >>> from ray.rllib.algorithms.ddpg.apex import ApexDDPGConfig
+        >>> from ray import tune
+        >>> config = ApexDDPGConfig()
+        >>> # Print out some default values.
+        >>> print(config.lr) # doctest: +SKIP
+        0.0004
+        >>> # Update the config object.
+        >>> config.training(lr=tune.grid_search([0.001, 0.0001]))
+        >>> # Set the config object's env.
+        >>> config.environment(env="Pendulum-v1")
+        >>> # Use to_dict() to get the old-style python config dict
+        >>> # when running with tune.
+        >>> tune.run(
+        ...     "APEX_DDPG",
+        ...     stop={"episode_reward_mean": 200},
+        ...     config=config.to_dict(),
+        ... )
+    """
 
-APEX_DDPG_DEFAULT_CONFIG = DDPGTrainer.merge_trainer_configs(
-    DDPGConfig().to_dict(),  # see also the options in ddpg.py, which are also supported
-    {
-        "optimizer": {
+    def __init__(self, trainer_class=None):
+        """Initializes a DDPGConfig instance."""
+        super().__init__(trainer_class=trainer_class or ApexDDPGTrainer)
+
+        # fmt: off
+        # __sphinx_doc_begin__
+        # ApexDDPG-specific settings.
+        self.optimizer = {
             "max_weight_sync_delay": 400,
             "num_replay_buffer_shards": 4,
             "debug": False,
-        },
-        "exploration_config": {"type": "PerWorkerOrnsteinUhlenbeckNoise"},
-        "n_step": 3,
-        "num_gpus": 0,
-        "num_workers": 32,
-        "replay_buffer_config": {
+        }
+        # max number of inflight requests to each sampling worker
+        # see the AsyncRequestsManager class for more details
+        # Tuning these values is important when running experimens with large sample
+        # batches. If the sample batches are large in size, then there is the risk that
```

Comment:
do we need to have this here? should be covered below, right?

Reply:
True!

Reply:
removed.